### PR TITLE
changes with the comments

### PR DESCRIPTION
**Pull request by EXR-85**

### CODE REVIEW

File: src/main/java/com/mhcure/logmerge/filewriter/MhFileWriter.java [hunk 1]
```diff
@@ -48,21 +47,21 @@ public void writeToFile(TreeMap<Long, String> fileContentsTreeMap) throws IOExce
         double countLines = 0;
         // Display the TreeMap which is naturally sorted
         TreeMap<Long, String> sortedTreeMapWithFileLines = fileContentsTreeMap;
-        if (splitLogFilesIntoMultiple.equalsIgnoreCase(MhFileConstants.USER_PERMISSION_YES)) {
+        if (splitLogFilesIntoMultiple.equalsIgnoreCase(MhFileConstants.USER_PERMISSION_N0)) {
             saveInSingleFile(sortedTreeMapWithFileLines, bufSize);
             return;
         }
         MhFileAggregatorHelper.printToConsole(MhFileConstants.USER_PROMPT_SPACE);
         MhFileAggregatorHelper.printToConsole(MhMessagePropertiesFileReader.getMessage(MhMessageKeyEnum.ASK_MAXIMUM_LINES_IN_LOG_FILE.getKey()));
```
Comment (EXR-86) [inline src/main/java/com/mhcure/logmerge/filewriter/MhFileWriter.java]: As the numberOfLinesPerOutPutFile is accepting from properties file, no need of these line no 54 to 56 u can remove.

File: src/main/resources/messages.properties [hunk 2]
```diff
@@ -1,21 +1,27 @@
...
+com.mhcure.userInfo.message.totalTime.to.writeFiles=total time (in seconds) to write files = 
 com.mhcure.userInfo.message.validLocation=Invalid location. Pls specify a valid location and restart the application.
-com.mhcure.userPrompt.message.ask_total_maximum_number_of_lines_in_each_log_file=Please specify maximum number of lines in each file\nAny number less than 1 considered as single file
+com.mhcure.userPrompt.message.ask_total_maximum_number_of_lines_in_each_log_file=Please specify maximum number of lines in each file\
+  \nAny number less than 1 considered as single file
 com.mhcure.userPrompt.message.folderSpecified=log files folder specified is logFolderLocation\nIf the value is incorrect, update \
   the corresponding property in the MhFileAggregator.Properties file and restart the program.\
   \nTo merge all log files from this location, enter 'Y' to exit the program, press 'X'.
 com.mhcure.userPrompt.message.locationIs=log files location is logFolderLocation
+com.mhcure.userInfo.message.save.decrypted.files="If you wish to save decrypted files, please modify the \
```
Comment (EXR-86) [inline src/main/resources/messages.properties]: remove those double quotes and '+' symbols present, just check and clean it up.

File: src/main/java/com/mhcure/logmerge/filewriter/MhFileWriter.java [hunk 3]
```diff
@@ -48,21 +47,21 @@ public void writeToFile(TreeMap<Long, String> fileContentsTreeMap) throws IOExce
...
         MhFileAggregatorHelper.printToConsole(MhFileConstants.USER_PROMPT_SPACE);
-        if (numberOfLinesPerOutputfiles <= 1000) {
```
Comment (EXR-86) [inline src/main/java/com/mhcure/logmerge/filewriter/MhFileWriter.java:58]: variable name numberOfLinesPerOutputfiles --> make it in camel case (numberOfLinesPerOutPutFiles

File: src/main/java/com/mhcure/logmerge/filereader/MhFileReader.java [hunk 4]
```diff
@@ -103,7 +103,7 @@ public Map<Long, String> readFileUsingBufferedReader(String fileName) throws Exc
         boolean isEncryptedFile = false;
```
Comment (EXR-86) [inline src/main/java/com/mhcure/logmerge/filereader/MhFileReader.java:103]: have a look at line no 88 u can move that hard coded string to constants.

File: src/main/java/com/mhcure/logmerge/filereader/MhFileReader.java [hunk 5]
```diff
@@ -103,7 +103,7 @@ public Map<Long, String> readFileUsingBufferedReader(String fileName) throws Exc
...
         Cipher cipherObject = null;
         if (!logFile.isFile()) {
-            System.out.println(fileName + MhMessagePropertiesFileReader.getMessage(MhMessageKeyEnum.IGNORE_INVALID_FILE.getKey()));
+            MhFileAggregatorHelper.printToConsole(fileName + MhMessagePropertiesFileReader.getMessage(MhMessageKeyEnum.IGNORE_INVALID_FILE.getKey()));
             return fileContentsMap;
         }
```
Comment (EXR-86) [inline src/main/java/com/mhcure/logmerge/filereader/MhFileReader.java:108]: at line no 119 the variable lineCounter is unused .

File: src/main/java/com/mhcure/logmerge/filereader/MhFileReader.java [hunk 6]
```diff
@@ -103,7 +103,7 @@ public Map<Long, String> readFileUsingBufferedReader(String fileName) throws Exc
...
         logFileReader = new FileReader(logFile);
```
Comment (EXR-86) [inline src/main/java/com/mhcure/logmerge/filereader/MhFileReader.java]: line no 171 no need of variable declaration
final String ALGO = MhFileConstants.ALGO
u can directly use that constants

File: src/main/java/com/mhcure/logmerge/LogAggregatorApplication.java [hunk 7]
```diff
@@ -64,10 +65,10 @@ public int getExitCode() {
...
-            if (isMergingRequired()) {
-                programStartTime = System.currentTimeMillis();
-                performLogAggregation();
-            }
+        if (isMergingRequired()) {
+            programStartTime = System.currentTimeMillis();
+            performLogAggregation();
+        }
         long programEndTime = System.currentTimeMillis();
         double programTimeInseconds = getTimeDiffInSeconds(programEndTime, programStartTime);
```
Comment (EXR-86) [inline src/main/java/com/mhcure/logmerge/LogAggregatorApplication.java]: change variable name to programTimeInSeconds

File: src/main/java/com/mhcure/logmerge/LogAggregatorApplication.java [hunk 8]
```diff
@@ -90,7 +91,7 @@ private boolean isMergingRequired() {
                 MhFileAggregatorHelper.printToConsole(MhMessagePropertiesFileReader
```
Comment (EXR-86) [inline src/main/java/com/mhcure/logmerge/LogAggregatorApplication.java:94]: line no 85 u can move hardcoded string to constants. similarly do it for all files.

File: src/main/java/com/mhcure/logmerge/LogAggregatorApplication.java [hunk 9]
```diff
@@ -123,24 +125,25 @@ private void performLogAggregation() throws IOException {
...
-                        getMessage(MhMessageKeyEnum.MESSAGE_INVALID_FILE_LOCATION.getKey()));
-                MhFileAggregatorHelper.printToConsole(MhFileConstants.LINE_SEPARATOR);
-                return;
+            MhFileAggregatorHelper.printToConsole(MhMessagePropertiesFileReader.
+                    getMessage(MhMessageKeyEnum.MESSAGE_INVALID_FILE_LOCATION.getKey()));
+            MhFileAggregatorHelper.printToConsole(MhFileConstants.LINE_SEPARATOR);
+            return;
         }
         List<String> fileContentsList = new ArrayList(fileContentsMap.values());
         long fileWriteStartTime = System.currentTimeMillis();
```
Comment (EXR-86) [inline src/main/java/com/mhcure/logmerge/LogAggregatorApplication.java:149]: line no 148 List declaration is unused.

**Review by EXR-86 — CHANGES_REQUESTED** [review]
Go through the comments and check which all applicable.

File: src/main/java/com/mhcure/logmerge/LogAggregatorApplication.java [hunk 10]
```diff
@@ -124,15 +123,14 @@ private void performLogAggregation() throws IOException {
                     MhFileAggregatorHelper.printToConsole(MhMessagePropertiesFileReader.getMessage(MhMessageKeyEnum.IGNORE_MERGING.getKey()));
                 }
                 MhFileAggregatorHelper.printToConsole(MhMessagePropertiesFileReader.getMessage(MhMessageKeyEnum.MESSAGE_PROCESSING_FILE
```
Comment (EXR-86) [inline src/main/java/com/mhcure/logmerge/LogAggregatorApplication.java:128]: Correct  prompt statement it shows "finished processing 1/totalfiles files".

File: src/main/java/com/mhcure/logmerge/LogAggregatorApplication.java
Comment (EXR-86) [inline src/main/java/com/mhcure/logmerge/LogAggregatorApplication.java:1]: If the decrypted file location is invalid then it is showing exception. Try solving this scenario.
even if the user is changing the properties to y for saving decrypted files then also it is prompting "If you wish to save decrypted files, please modify the relevant property in MhFileAggregator. Properties as the program currently does not retain decrypted files" try prompting something relevant to it

Comment (EXR-86) [inline src/main/java/com/mhcure/logmerge/LogAggregatorApplication.java:1]: If the log files out put location is incorrect then also exception is rising. So try out a logic to handle these exception and prompt a relevant message to user

Comment (EXR-86) [inline src/main/java/com/mhcure/logmerge/LogAggregatorApplication.java:1]: If the encryption key is invalid then InvalidKeyException is rising. try out to handle this and provide a relevant message to give correct encryption key

Comment (EXR-86) [inline src/main/java/com/mhcure/logmerge/LogAggregatorApplication.java:1]: Try checking file extension (which is relevant for this tool ). So if it is .png or .jpg or something different which is not relevant, the tool should not execute . It should ask for relevant files.

**Review by EXR-86 — COMMENTED** [review]
Try out solving these new comments.

**Review by EXR-86 — CHANGES_REQUESTED** [review]
Try out solving these new comments.

File: src/main/java/com/mhcure/logmerge/filewriter/MhFileWriter.java (hunk 1 above)
Comment (EXR-85) [reply src/main/java/com/mhcure/logmerge/filewriter/MhFileWriter.java]: Removed

File: src/main/resources/messages.properties (hunk 2 above)
Comment (EXR-85) [reply src/main/resources/messages.properties]: removed

File: src/main/java/com/mhcure/logmerge/filewriter/MhFileWriter.java (hunk 3 above)
Comment (EXR-85) [reply src/main/java/com/mhcure/logmerge/filewriter/MhFileWriter.java:58]: done

File: src/main/java/com/mhcure/logmerge/filereader/MhFileReader.java (hunk 4 above)
Comment (EXR-85) [reply src/main/java/com/mhcure/logmerge/filereader/MhFileReader.java:103]: done

File: src/main/java/com/mhcure/logmerge/filereader/MhFileReader.java (hunk 5 above)
Comment (EXR-85) [reply src/main/java/com/mhcure/logmerge/filereader/MhFileReader.java:108]: removed

File: src/main/java/com/mhcure/logmerge/filereader/MhFileReader.java (hunk 4 above)
Comment (EXR-85) [reply src/main/java/com/mhcure/logmerge/filereader/MhFileReader.java:103]: done

File: src/main/java/com/mhcure/logmerge/filereader/MhFileReader.java (hunk 6 above)
Comment (EXR-85) [reply src/main/java/com/mhcure/logmerge/filereader/MhFileReader.java]: I changed according to the comment

File: src/main/java/com/mhcure/logmerge/LogAggregatorApplication.java (hunk 7 above)
Comment (EXR-85) [reply src/main/java/com/mhcure/logmerge/LogAggregatorApplication.java]: done

File: src/main/java/com/mhcure/logmerge/LogAggregatorApplication.java (hunk 8 above)
Comment (EXR-85) [reply src/main/java/com/mhcure/logmerge/LogAggregatorApplication.java:94]: Done

File: src/main/java/com/mhcure/logmerge/LogAggregatorApplication.java (hunk 9 above)
Comment (EXR-85) [reply src/main/java/com/mhcure/logmerge/LogAggregatorApplication.java:149]: done

**Pull request merged**